src/Octoshift/Commands/AbortMigration/AbortMigrationCommandHandler.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Threading.Tasks;
+using OctoshiftCLI.Services;
+
+
+namespace OctoshiftCLI.Commands.AbortMigration;
+
+public class AbortMigrationCommandHandler : ICommandHandler<AbortMigrationCommandArgs>
+{
+
+    private readonly OctoLogger _log;
+    private readonly GithubApi _githubApi;
+
+    public AbortMigrationCommandHandler(OctoLogger log, GithubApi githubApi)
+    {
+        _log = log;
+        _githubApi = githubApi;
+    }
+
+    public async Task Handle(AbortMigrationCommandArgs args)
+    {
+        if (args is null)
+        {
+            throw new ArgumentNullException(nameof(args));
+        }
+
+        await AbortRepositoryMigration(args.MigrationId);
+    }
+
+    private async Task AbortRepositoryMigration(string migrationId)
+    {
+        var abortionState = await _githubApi.AbortMigration(migrationId);
+
+        if (!abortionState)
+        {
+            _log.LogError($"Failed to abort migration {migrationId}");
```

Comment:
I agree but this needs to come from the backend API so I guess we need another ticket to tweak the API.

Reply:
In that case the GQL endpoint will return an error and then the CLI will show a generic error in non verbose mode and log the error anyways. And of course the error message is getting returned by the backend. I think this default behavior is good enough unless we have a strong case to tweak the api call method in the CLI.

Reply:
We can create a ticket to tweak the backend API to not return and error for those kind of failures (e.g. migration id is not found, etc.) and instead returns a payload that has a success and failureReason that will be set in case of a failure.

Reply:
While the error message would be surfaced as is I think its a nice idea to surface explicit error messages for each of these scenarios! Created [this](https://github.ghe.com/github/octoshift/issues/7842) WI for the refactor

Reply:
> We can create a ticket to tweak the backend API to not return an error for those kind of failures (e.g. migration id is not found, etc.) and instead returns a payload that has a `success` and `failureReason` that will be set in case of a failure.

We'll need to be careful about doing that, as theoretically it's a breaking change to the current behaviour (albeit for an API we only just launched). But we can discuss that in a separate follow-up issue.